File: content/problems/1663-smallest-string-with-a-given-numeric-value/solutions/right-to-left-greedy/solution.cpp
class Solution {
    public: string getSmallestString(int n,int k){
        string letters(n,'a');
        int remaining=k-n;
        for(int index=n-1;index>=0&&remaining;--index){
            int add=min(25,remaining);
            letters[index]+=add;
            remaining-=add;
        }
        return letters;
    }
};

File: content/problems/1663-smallest-string-with-a-given-numeric-value/solutions/right-to-left-greedy/Solution.java
class Solution {
    public String getSmallestString(int n,int k){
        char[] letters=new char[n];
        Arrays.fill(letters,'a');
        int remaining=k-n;
        for(int index=n-1;index>=0&&remaining>0;index--){
            int add=Math.min(25,remaining);
            letters[index]+=add;
            remaining-=add;
        }
        return new String(letters);
    }
}
